### ISSUE TITLE
FULL Non-Merged (+bios,+device): handling "samename" rom conflicts

### ISSUE THREAD

**Issue by cinnamonmatexfce** [optyfr/JRomManager#133]
How could JRomManager cope with rom(s) having same exact name?
Till now only this peculiar case happened to me (so, not really a big problem after all...).
M. A. M. E. romset is _FULL Non-Merged (including bioses and devices)_.
The conflict, the only one so far, is for the `317-unknown.pic` rom, present in these machine(s):
- `scg05nt` (game, parent)
- `scg05ntb` (game, clone)
- `segadimm` (device)

Rom has same name and same size, but **differs in CRC and SHA1**.
![317-unknown pic](https://user-images.githubusercontent.com/65762930/139838507-c7e1fd26-658a-48dc-a05a-a7da3f406a39.jpg)

Infos down here (taken from the official M. A. M. E. XML):
```
SCG05NT
<machine name="scg05nt" sourcefile="chihiro.cpp" romof="chihiro">
		<description>Sega Golf Club Network Pro Tour 2005 (Rev C) (GDX-0010C)</description>
		<year>2004</year>
		<manufacturer>Sega</manufacturer>
		<biosset name="bios0" description="Chihiro Bios"/>
		<rom name="317-unknown.pic" size="16384" crc="36858860" sha1="b36a0c10b614fdeb7dcd94f4898efc24e5db896a" region="pic" offset="0"/>

###

SCG05NTB
<machine name="scg05ntb" sourcefile="chihiro.cpp" cloneof="scg05nt" romof="scg05nt">
		<description>Sega Golf Club Network Pro Tour 2005 (Rev B) (GDX-0010B)</description>
		<year>2004</year>
		<manufacturer>Sega</manufacturer>
		<biosset name="bios0" description="Chihiro Bios"/>
		<rom name="317-unknown.pic" merge="317-unknown.pic" size="16384" crc="36858860" sha1="b36a0c10b614fdeb7dcd94f4898efc24e5db896a" region="pic" offset="0"/>

###

SEGADIMM
<machine name="segadimm" sourcefile="src/mame/machine/naomigd.cpp" isdevice="yes" runnable="no">
		<description>Sega DIMM Board</description>
		<biosset name="fpr-23489c.ic14" description="Bios 0"/>
		<biosset name="203_203.bin" description="Bios 1"/>
		<biosset name="fpr23718.ic36" description="Bios 2"/>
		<biosset name="213_203.bin" description="Bios 3"/>
		<biosset name="217_203.bin" description="Bios 4"/>
		<biosset name="fpr23905.ic36" description="Bios 5"/>
		<biosset name="317_312.bin" description="Bios 6"/>
		<biosset name="401_203.bin" description="Bios 7"/>
		<rom name="317-unknown.pic" size="16384" crc="7dc07733" sha1="b223dc44718fa71e7b420c3b44ce4ab961445461" region="segadimm" offset="0"/>
```

Would be doable prioritizing the selection of game's rom over the device's one? Does this make sense?

Close if unworthy ;)

**Comment by optyfr** [comment]
@cinnamonmatexfce I should look into that closely to understand what happens really... was that the latest MAME?

**Comment by cinnamonmatexfce** [comment]
To be fair it happened always to what I remember since I first time I started using JRomManager. I never report (till now) 'cause it wasn't causing too much troubles to me (these 2 sets are the only ones problematics, and every time I ended up fixing them by hand).

**Comment by optyfr** [comment]
OK, in the mean time the partial fix bug reappeared, I will have to fix that critical bug first before continuing

**Comment by optyfr** [comment]
A solution have been implemented, doing some extra tests...

**Comment by optyfr** [comment]
Considered as fixed for 2.5.1
